recipes/sysctl.rb

```diff
@@ -20,6 +20,13 @@
 # limitations under the License.
 #
 
+# Do not run the sysctl recipe on LXD/LXC containers as the values cannot be
+# changed from within a container.
```

Comment:
Not that I am saying this is a good idea but I am often forced to run privileged containers in the context of infrastructure CI. In order to really test that services start they often require being able to access the hosts systemd/upstart or other resources not normally available. I think this could be in addition to checking those ohai values we should have a flag to control the default. You might need to do it just for your CI masters and executors but other nodes it would make sense to otherwise take the sane default of not assuming not privileged. Another option would be to try attempt running something that could only be done if privileged. Something along these lines?
```bash
ip link add dummy0 type dummy >/dev/null
if [[ $? -eq 0 ]]; then
    PRIVILEGED=true
    # clean the dummy0 link
    ip link delete dummy0 >/dev/null
else
    PRIVILEGED=false
fi
```

Reply:
The problem with using a privileged container is that when you modify sysctl settings in the container, you are actually changing them on the host. This probably isn't what most people would want or expect it to do.